src/components/models/CartModel.ts

```diff
@@ -0,0 +1,232 @@
+        } else {
+            this.productsInCart = [];
+        }
+
+        if (localStorage.getItem('activatedPromocodes')) {
+            this.activatedPromocodes = JSON.parse(localStorage.getItem('activatedPromocodes') || '{}');
+        } else {
+            this.activatedPromocodes = [];
+        }
+
+        this.totalSumWithDiscount = this.checkTotalSumWithDiscountInLocalStorage();
+
+        this.enteredPromocodes = this.setEnteredPromocodesAfterReload();
+    }
+
+    private checkTotalSumInLocalStorage() {
+        const totalSumInLocalStorage = window.localStorage.getItem('totalSum');
+        if (totalSumInLocalStorage) {
+            this.totalSum = Number(totalSumInLocalStorage);
+        } else {
+            this.totalSum = 0;
+        }
+        return this.totalSum;
+    }
+
+    private checkTotalSumWithDiscountInLocalStorage() {
+        const totalSumWithDiscountInLocalStorage = window.localStorage.getItem('totalSumWithDiscount');
+        if (totalSumWithDiscountInLocalStorage) {
+            this.totalSumWithDiscount = Number(totalSumWithDiscountInLocalStorage);
+        } else {
+            this.totalSumWithDiscount = this.totalSum;
+        }
+        return this.totalSumWithDiscount;
+    }
+
+    private checkProductQuantityInLocalStorage() {
+        const productQuantityInLocalStorage = localStorage.getItem('productsQuantity');
+        if (productQuantityInLocalStorage) {
+            this.productsQuantity = Number(productQuantityInLocalStorage);
+        } else {
+            this.productsQuantity = 0;
+        }
+        return this.productsQuantity;
+    }
+
+    addProduct(product: Product) {
+        const isInCart = this.checkProductInCart(product.id);
+        if (!isInCart) {
+            this.productsInCart.push(product);
+            product.inCart = 1;
+            product.sum = product.price;
+            this.productsQuantity += product.inCart;
+            this.totalSum = Number((this.totalSum += product.inCart * product.price).toFixed(2));
+            this.appController.addProductToCart(product);
+            this.calculateAddingSumDiscoutDelete();
+            this.appController.cartView.updateDiscountInfo();
+            return this.productsInCart;
+        }
+    }
+
+    deleteProduct(product: Product) {
+        const isInCart = this.checkProductInCart(product.id);
+        if (isInCart) {
+            const deletedIndex = this.productsInCart.indexOf(product);
+            this.productsInCart.splice(deletedIndex, 1);
+            if (product.inCart) {
+                this.productsQuantity -= product.inCart;
+                this.totalSum = Number((this.totalSum -= product.inCart * product.price).toFixed(2));
+                product.inCart = 0;
+                this.appController.cartView.totalPerProduct = 0;
+                const currentTotalPercentDiscount = this.getCurrentTotalPercentDiscount();
+                this.totalSumWithDiscount = Number(
```

Comment:
You are using the same calculation in several different places. I suggest to put them into method.

Reply:
Fix it